src/adapter/src/rbac.rs

```diff
@@ -146,33 +207,12 @@ pub fn check_plan(
     target_cluster_id: Option<ClusterId>,
     resolved_ids: &ResolvedIds,
 ) -> Result<(), AdapterError> {
-    let current_role_id = session.current_role_id();
-    if catalog.try_get_role(current_role_id).is_none() {
-        // PostgreSQL allows users that have their role dropped to perform some actions,
-        // such as `SET ROLE` and certain `SELECT` queries. We haven't implemented
-        // `SET ROLE` and feel it's safer to force to user to re-authenticate if their
-        // role is dropped.
-        return Err(AdapterError::ConcurrentRoleDrop(current_role_id.clone()));
-    };
+    rbac_preamble!(catalog, session);
```

Comment:
The preamble is callled in check_item_usage, so this calls it twice. Could we remove this?